nixos/modules/services/misc/conduit.nix

```diff
@@ -0,0 +1,118 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.conduit;
+
+  format = pkgs.formats.toml {};
+  namespacedSettings.global = cfg.settings;
+  configFile = format.generate "conduit.toml" namespacedSettings;
+in
+  {
+    options.services.conduit = {
+      enable = mkEnableOption "conduit";
+      iHaveReadTheFederationWarning = mkOption {
+        type = types.bool;
+        default = false;
+        example = true;
+        description = ''
+          Upstream documentation contains the following warning:
+          You should not join Matrix rooms without asking the admins first.
+          We do not know whether Conduit is safe for general use yet, so you
+          should assume there is some chance that it breaks rooms permanently
+          for all participating users. We are not aware of such a bug today,
+          but we would like to do more testing.
+
+          This option will be removed upon removal of this warning.
+        '';
+      };
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.conduit;
+        defaultText = "pkgs.conduit";
+        example = "pkgs.conduit";
+        description = ''
+          Package of the conduit matrix server to use.
+        '';
+      };
+
+      settings = mkOption {
```

Comment:
I think it is a good pattern to add explicit configuration options within the submodule for required options or those that will almost always be set. `server_name` is a strong contender for this, maybe also `port`.

Reply:
I always find that more confusing than just having it all together at one level. Unnecessary complexity imho.